From 29b8073e6c773cdee5d7133e7eabdc9bec4d0a18 Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Wed, 26 Sep 2018 11:39:48 +0200
Subject: [PATCH] Merge pull request #209 in SW_UGW/linux from
 bugfix/PPA_SYS-565-falcon-mountain-qos_hal_drv-does-not-load-2 to master

* commit '9a447939cf70f7e626a576f700de51bf0fae0489':
  PPA_SYS-565: Add some callback implementations again.
---
 drivers/net/ethernet/lantiq/ppa/ppa_drv_wrapper.c | 24 +++++++++++++++++++++++
 drivers/net/ethernet/lantiq/ppa/ppa_hook.c        |  3 +++
 2 files changed, 27 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/ppa/ppa_drv_wrapper.c b/drivers/net/ethernet/lantiq/ppa/ppa_drv_wrapper.c
index 9d99ed0d0b9a..4d2ccce4ae5a 100644
--- a/drivers/net/ethernet/lantiq/ppa/ppa_drv_wrapper.c
+++ b/drivers/net/ethernet/lantiq/ppa/ppa_drv_wrapper.c
@@ -135,6 +135,30 @@ void (*ppa_session_mc_destroy_tmplbuf_hook)(void* sessionAction) = NULL;
 EXPORT_SYMBOL(ppa_session_mc_destroy_tmplbuf_hook);
 #endif
 
+int32_t (*qos_hal_get_csum_ol_mib_hook_fn)(
+		struct qos_hal_qos_stats *csum_mib,
+		uint32_t flag) = NULL;
+EXPORT_SYMBOL(qos_hal_get_csum_ol_mib_hook_fn);
+
+int32_t (*qos_hal_clear_csum_ol_mib_hook_fn)(
+		uint32_t flag) = NULL;
+EXPORT_SYMBOL(qos_hal_clear_csum_ol_mib_hook_fn);
+
+int32_t(*qos_hal_get_qos_mib_hook_fn)(
+		struct net_device *netdev,
+		dp_subif_t *subif_id,
+		int32_t queueid,
+		struct qos_hal_qos_stats *qos_mib,
+		uint32_t flag) = NULL;
+EXPORT_SYMBOL(qos_hal_get_qos_mib_hook_fn);
+
+int32_t (*qos_hal_clear_qos_mib_hook_fn)(
+		struct net_device *netdev,
+		dp_subif_t *subif_id,
+		int32_t queueid,
+		uint32_t flag) = NULL;
+EXPORT_SYMBOL(qos_hal_clear_qos_mib_hook_fn);
+
 /*	others:: these hook will be set in PPE datapath driver	*/
 int (*ppa_drv_get_dslwan_qid_with_vcc_hook)(struct atm_vcc *vcc) = NULL;
 int (*ppa_drv_get_netif_qid_with_pkt_hook)(PPA_SKBUF *skb, void *arg, int is_atm_vcc) = NULL;
diff --git a/drivers/net/ethernet/lantiq/ppa/ppa_hook.c b/drivers/net/ethernet/lantiq/ppa/ppa_hook.c
index 5c5f12b9c2ed..d83a357bcf82 100644
--- a/drivers/net/ethernet/lantiq/ppa/ppa_hook.c
+++ b/drivers/net/ethernet/lantiq/ppa/ppa_hook.c
@@ -390,6 +390,9 @@ int32_t (*ppa_hook_reset_qos_wfq)(uint32_t portid, uint32_t queueid, uint32_t fl
 
 int32_t (*qos_mgr_hook_setup_tc)(struct net_device *dev, u32 handle, __be16 protocol, struct tc_to_netdev *tc) = NULL;
 
+int32_t (*qos_mgr_hook_setup_tc_ext)(struct net_device *dev, u32 handle, __be16 protocol, struct tc_to_netdev *tc, int32_t deq_idx, int32_t port_id) = NULL;
+EXPORT_SYMBOL(qos_mgr_hook_setup_tc_ext);
+
 #ifdef CONFIG_INTEL_IPQOS_MARK_SKBPRIO
 /*
  * Function to mark priority based on specific criteria
